NS_HTTP_PROVIDERS angular rc4 with nativescript-angular@0.3.1 still problem

If you just import `NS_HTTP_PROVIDERS` and set on your bootstrap, you won't pass through the Typescript compiler and will get this error:

```bash
Found peer TypeScript 1.8.10
node_modules/nativescript-angular/http/ns-http.d.ts(8,22): error TS2415: Class 'NSHttp' incorrectly extends base class 'Http'.
  Types of property 'get' are incompatible.
    Type '(url: string, options?: RequestOptionsArgs) => Observable<any>' is not assignable to type '(url: string, options?: RequestOptionsArgs) => Observable<Response>'.
      Type 'Observable<any>' is not assignable to type 'Observable<Response>'.
        Property 'source' is protected but type 'Observable<T>' is not a class derived from 'Observable<T>'.
```

To check this bug, follow these steps:

```bash
tns create myapp --template tns-template-hello-world-ng
```

Then, import NS_HTTP_PROVIDERS:

```javascript
import {NS_HTTP_PROVIDERS} from 'nativescript-angular/http';
nativeScriptBootstrap(AppComponent, [NS_HTTP_PROVIDERS])
```

So try add a platform and run:

```
tns platform add ios
tns run ios --emulator
```

**Temporary solution**
> If you've set `compilerOptions` to false on `tsconfig.json`, the tsc will ignore it, and your app will run great. (Even if you use Http service).


I believe something related to the declaration on `node_modules/nativescript-angular/http/ns-http.d.ts` that doesn't follow with the angular rc-4.

--- a/nativescript-angular/http/ns-http.ts
+++ b/nativescript-angular/http/ns-http.ts
@@ -1,5 +1,5 @@
 import {Injectable} from '@angular/core';
-import {HTTP_PROVIDERS, Http, XHRBackend, XHRConnection, ConnectionBackend, RequestOptions, RequestOptionsArgs, ResponseOptions, ResponseType, Response, Request, BrowserXhr, XSRFStrategy} from '@angular/http';
+import {HTTP_PROVIDERS, Http, XHRBackend, ConnectionBackend, RequestOptions, RequestOptionsArgs, ResponseOptions, ResponseType, Response, XSRFStrategy} from '@angular/http';
 import {Observable} from 'rxjs/Observable';
 import 'rxjs/add/observable/fromPromise';
 import {NSFileSystem} from '../file-system/ns-file-system';
@@ -8,7 +8,7 @@ export class NSXSRFStrategy {
   public configureRequest(req: any) {
     // noop
   }
-}  
+}
 
 @Injectable()
 export class NSHttp extends Http {
@@ -20,12 +20,12 @@ export class NSHttp extends Http {
    * Performs a request with `get` http method.
    * Uses a local file if `~/` resource is requested.
    */
-  get(url: string, options?: RequestOptionsArgs): Observable<Response | any> {
+  get(url: string, options?: RequestOptionsArgs): Observable<Response> {
     if (url.indexOf('~') === 0 || url.indexOf('/') === 0) {
       // normalize url
       url = url.replace('~', '').replace('/', '');
       // request from local app resources
-      return Observable.fromPromise(new Promise((resolve, reject) => {
+      return Observable.fromPromise<Response>(new Promise((resolve, reject) => {
         let app = this.nsFileSystem.currentApp();
         let localFile = app.getFile(url);
         if (localFile) {
@@ -58,7 +58,8 @@ export const NS_HTTP_PROVIDERS: any[] = [
   HTTP_PROVIDERS,
   { provide: XSRFStrategy, useValue: new NSXSRFStrategy() },
   NSFileSystem,
-  { provide: Http, useFactory: (backend, options, nsFileSystem) => {
+  {
+    provide: Http, useFactory: (backend, options, nsFileSystem) => {
       return new NSHttp(backend, options, nsFileSystem);
     }, deps: [XHRBackend, RequestOptions, NSFileSystem]
   }

